ambry-tools/src/integration-test/java/com.github.ambry/account/AccountUpdateToolTest.java

```diff
@@ -0,0 +1,226 @@
+    idConflictAccounts.add(new AccountBuilder((short) 1, "account2", Account.AccountStatus.INACTIVE, null).build());
+    try {
+      createOrUpdateAccountsAndWait(idConflictAccounts);
+      fail("Should have thrown.");
+    } catch (IllegalArgumentException e) {
+      // expected
+    }
+    assertEquals("Wrong number of accounts in accountService", 0, accountService.getAllAccounts().size());
+
+    // name conflict
+    Collection<Account> nameConflictAccounts = new ArrayList<>();
+    nameConflictAccounts.add(new AccountBuilder((short) 1, "account1", Account.AccountStatus.INACTIVE, null).build());
+    nameConflictAccounts.add(new AccountBuilder((short) 2, "account1", Account.AccountStatus.INACTIVE, null).build());
+    try {
+      createOrUpdateAccountsAndWait(nameConflictAccounts);
+      fail("Should have thrown.");
+    } catch (IllegalArgumentException e) {
+      // expected
+    }
+    assertEquals("Wrong number of accounts in accountService", 0, accountService.getAllAccounts().size());
+  }
+
+  /**
+   * Tests creating/updating {@link Account}s using bad file containing no valid json array.
+   * @throws Exception Any unexpected exception.
+   */
+  @Test
+  public void testBadJsonFile() throws Exception {
+    String badJsonFile = tempDirPath + File.separator + "badJsonFile.json";
+    writeStringToFile("Invalid json string", badJsonFile);
+    try {
+      AccountUpdater.createOrUpdate(badJsonFile, ZK_SERVER_ADDRESS, HELIX_STORE_ROOT_PATH, null, null);
+      fail("Should have thrown.");
+    } catch (JSONException e) {
+      // expected
+    }
+    Thread.sleep(100);
+    assertEquals("Wrong number of accounts in accountService", 0, accountService.getAllAccounts().size());
+  }
+
+  /**
+   * Writes a collection of {@link Account}s into a file. The file contains and only contains a {@link JSONArray}, where
+   * each element is the json string of the {@link Account}.
+   * @param accounts The collection of {@link Account}s to write to the file.
+   * @param filePath The path of the file to write.
+   * @throws Exception Any unexpected exception.
+   */
+  private static void writeAccountsToFile(Collection<Account> accounts, String filePath) throws Exception {
+    JSONArray accountArray = new JSONArray();
+    for (Account account : accounts) {
+      accountArray.put(account.toJson());
+    }
+    writeJsonArrayToFile(accountArray, filePath);
+  }
+
+  /**
+   * Creates or updates a collection of {@link Account}s to {@code ZooKeeper} server.
+   * @param accounts The collection of {@link Account}s to create or update.
+   * @return The file name if the operation is successful; {@code null} otherwise.
+   * @throws Exception Any unexpected exception.
+   */
+  private static String createOrUpdateAccountsAndWait(Collection<Account> accounts) throws Exception {
+    String jsonFilePath = tempDirPath + File.separator + UUID.randomUUID().toString() + ".json";
+    writeAccountsToFile(accounts, jsonFilePath);
```

Comment:
I thought the `getTempDir` method should do the job to delete the root level dir, right?